sql/analyzer/resolve_columns.go

```diff
@@ -837,6 +837,9 @@ func indexColumns(_ *sql.Context, _ *Analyzer, n sql.Node, scope *Scope) (map[ta
 	// Index the columns in the outer scope, outer to inner. This means inner scope columns will overwrite the outer
 	// ones of the same name. This matches the MySQL scope precedence rules.
 	indexSchema(scope.Schema())
+	if !scope.IsEmpty() && len(columns) == 0 {
+		return nil, nil
+	}
```

Comment:
Might be nice to add a comment here explaining the issue. I had to think through this for a little bit and I already had some context on the issue. Or... maybe even just throwing this into a function called `outerScopesResolved()` or something would be self documenting enough?